tests/plots/test_index.py

```diff
@@ -8,126 +8,180 @@
-            "filter_col": ["X", "Y", "X", "Y", "X", "Y", "X", "Y", "X", "Y", "X", "Y"],
-            "value_col": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
+            "subgroup": ["X", "X", "X", "Y", "Y", "Y"],
+            "category": ["A", "A", "B", "B", "C", "C"],
+            "value": [10, 20, 30, 40, 50, 60],
         },
     )
-    expected_output = pd.DataFrame(
+
+    result = get_indexes(
+        df,
+        value_to_index="A",
+        index_col="category",
+        value_col="value",
+        group_col="category",
+        index_subgroup_col="subgroup",
+    )
+    assert isinstance(result, pd.DataFrame)
+    assert "category" in result.columns
+    assert "index" in result.columns
+    assert not result.empty
+
+
+def test_get_indexes_invalid_agg_func():
+    """Test get_indexes function with an invalid aggregation function."""
+    df = pd.DataFrame(
         {
-            "group_col2": ["D", "D", "D", "E", "E", "E"],
-            "group_col1": ["A", "B", "C", "A", "B", "C"],
-            "index": [77.77777778, 100, 106.0606, 98.51851852, 100, 100.9661836],
+            "category": ["A", "B", "C"],
+            "value": [10, 20, 30],
         },
     )
-    output = get_indexes(
-        df=df,
-        index_col="group_col1",
-        index_subgroup_col="group_col2",
-        df_index_filter=df["filter_col"] == "X",
-        value_col="value_col",
+
+    with pytest.raises(ValueError, match="Unsupported aggregation function."):
+        get_indexes(
+            df,
+            value_to_index="A",
+            index_col="category",
+            value_col="value",
+            group_col="category",
+            agg_func="invalid_func",
+        )
+
+
+def test_get_indexes_with_different_aggregations():
+    """Test get_indexes function with various aggregation functions."""
+    df = pd.DataFrame(
+        {
+            "category": ["A", "A", "B", "B", "C", "C"],
+            "value": [10, 20, 30, 40, 50, 60],
+        },
     )
-    pd.testing.assert_frame_equal(output, expected_output)
+
+    for agg in ["sum", "mean", "max", "min", "nunique"]:
+        result = get_indexes(
+            df,
+            value_to_index="A",
+            index_col="category",
+            value_col="value",
+            group_col="category",
+            agg_func=agg,
+        )
+        assert isinstance(result, pd.DataFrame)
+        assert "category" in result.columns
+        assert "index" in result.columns
+        assert not result.empty
 
 
 def test_get_indexes_with_offset():
-    """Test that the function works with an offset parameter."""
+    """Test get_indexes function with an offset value."""
+    df = pd.DataFrame(
+        {
+            "category": ["A", "B", "C"],
+            "value": [10, 20, 30],
+        },
+    )
+    result = get_indexes(
+        df,
+        value_to_index="A",
+        index_col="category",
+        value_col="value",
+        group_col="category",
+        offset=5,
```

Comment:
SInce this is tied to the value of OFFSET_THRESHOLD I think we should change this line.
```suggestion
        offset=-OFFSET_THRESHOLD,
```